src/transforms/sample/transform.rs

```diff
@@ -12,24 +12,86 @@ use crate::{
 use vector_lib::lookup::lookup_v2::OptionalValuePath;
 use vector_lib::lookup::OwnedTargetPath;
 
+/// Exists only for backwards compatability purposes so that the value of sample_rate_key is
+/// consistent after the internal implementation of the Sample class was modified to work in terms
+/// of percentages
+#[derive(Clone, Debug)]
+pub enum SampleMode {
+    Rate(u64, HashMap<Option<String>, u64>),
+    Ratio(f64, HashMap<Option<String>, f64>),
```

Comment:
These could be named for better clarity on what the using code is doing:
```suggestion
    Rate {
        rate: u64,
        counters: HashMap<Option<String>, u64>,
    },
    Ratio {
        ratio: f64,
        values: HashMap<Option<String>, f64>,
    },
```